### PR TITLE
WIP: GitHub.auth.for.rhdt dev.member

### PR DESCRIPTION
**Pull request by corinnekrych**

### CODE REVIEW

File: github-auth-hook.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    app.get('/api/admin/login', passport.authenticate('github'));
+    let context = process.env.TOGGLES_CONTEXT ? process.env.TOGGLES_CONTEXT : '';
+    app.get(
+        '/api/auth/callback',
+        passport.authenticate('github', {
+            failureRedirect: `${context}/api/admin/error-login`,
+        }),
+        (req, res) => {
+            // Successful authentication, redirect to your app.
+            res.redirect(`${context}/`);
```
Comment (corinnekrych) [inline github-auth-hook.js]: @aslakknutsen this redirect does not go through the proxy so it needs the context `/toggles/`

File: github-auth-hook.js [hunk 2]
```diff
@@ -14,22 +15,71 @@ passport.use(
...
             }
-            done(
-                null,
-                new User({
-                    name: profile.displayName,
-                    email: profile.emails[0].value,
-                })
+
+            let githubOrg = process.env.GITHUB_ORG ? process.env.GITHUB_ORG : 'rhdt-toggles-test';
+            let githubOrgTeam = process.env.GITHUB_ORG_TEAM ? process.env.GITHUB_ORG : 'toggles-admin-test';
```
Comment (corinnekrych) [inline github-auth-hook.js]: do you mean:
let githubOrgTeam = process.env.GITHUB_ORG_TEAM ? process.env.GITHUB_ORG_TEAM : 'toggles-admin-test';

Comment (xcoulon) [reply github-auth-hook.js]: ah right :/ let me fix that!

File: github-auth-hook.js [hunk 3]
```diff
@@ -34,15 +34,19 @@ passport.use(
                 function(error, response, body) {
                     if (error) {
                         console.error('access to GH org failed:', error);
-                        return done(error);
+                        return done(null, false, { message: error });
```
Comment (xcoulon) [inline github-auth-hook.js]: why did you need to change the args in this call to `done` ?

Comment (corinnekrych) [reply github-auth-hook.js]: bc I'd like to have a consistent way of displaying error in the field message sot that the UI can display them all in the same manner.

File: github-auth-hook.js [hunk 4]
```diff
@@ -22,7 +23,18 @@ passport.use(
                 // user can choose to not display any email, then use a default one as unleash required it
                 profile.emails.push(`${displayName}@unknown.com`);
             }
-            // Successful authentication, now check if the authenticated user is a member of the GH org/team
+            let user = new User({
+                name: profile.displayName,
+                email: profile.emails[0].value,
+            });
+
+            // Successful authentication, now check if the authenticated user is a member of the GH org/team, unless `dev mode` is enabled
```
Comment (corinnekrych) [inline github-auth-hook.js]: @xcoulon move this comment after dev-mode

**Comment by corinnekrych** [comment]
@aslakknutsen @xcoulon I think we got it working fine. I've tested 2 flows:
- user authenticates to GH but does not belong to GH org => pop-up to ask him to be added and sign-in button to retry
- user authenticates to GH  and belong to GH org/team => signed in
We miss the logout flow but it could be done on separate PR.

File: github-auth-hook.js (hunk 4 above)
Comment (xcoulon) [reply github-auth-hook.js]: the comment ends with `...unless `dev mode` is enabled.` so I think it makes sense to leave it before the `if (devmode) {..}` check.

**Comment by corinnekrych** [comment]
@aslakknutsen @xcoulon it's ready and proxy-friendy
see:
toggles_admin_proxy
